[optimistic-upgrade] Recommend GET for future Upgrade Tokens

bemasc opened the pull request:
Fixes #2738

royfielding commented:
I do not agree with this suggestion.

Upgrade is intended to support a not-yet-defined future. Limits placed upon it must be evaluated in terms of what will be excluded from future use, not what is currently being used in practice.

If you want to limit what can be sent in a body, do that instead. The entire premise of this document is that a client will send something incredibly stupid and uncontrolled in a request body while asking for an upgrade. Just forbid that and be done.

In any case, Upgrade with GET, HEAD, and OPTIONS has no such issues. Upgrade should also be possible to use with QUERY with any compliant server. If you can find a server that fails to process a valid request body and then proceeds to treat it as a pipelined request, that is a vulnerability in their code.

bemasc commented:
This PR originally assumed the deprecation of "TLS" in #2818.  I've removed that deprecation there, so other HTTP methods are necessarily allowed.

Due to that change, I've rewritten the recommendation here, removed the PR stacking, and removed the "draft PR" flag.
